src/evict/evict_file.c

```diff
@@ -93,16 +93,7 @@ __wt_evict_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
             WT_ERR(__wt_evict(session, ref, ref->state, WT_EVICT_CALL_CLOSING));
             break;
         case WT_SYNC_DISCARD:
-            /*
```

Comment:
This check is important - it notices if we are throwing away dirty data "accidentally", which generally corresponds to a data corruption. The `__wt_page_can_evict` check should still be correct. Does this change break the prior check here somehow?
It looks to me as though it'd still be OK?

Reply:
I put the check back in 7822d0e44.

(I think this was historically an exhaustive test, but that's no longer true. I agree it will still catch some cases where we discard dirty data we shouldn't discard, but as we continue to relax more constraints on what can be evicted in the future, it will become increasingly irrelevant.)

Reply:
Thanks and agreed.